rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -726,6 +740,26 @@ impl Controller {
         }
     }
 
+    async fn update_disabled_resources(&mut self) -> Result<()> {
+        settings::save(&self.advanced_settings).await?;
+
+        let Status::TunnelReady { resources } = &self.status else {
+            bail!("Tunnel is not ready");
+        };
```

Comment:
If you make another function on `Status` to retrieve the resources, you could move the `bail!` into there and just `?` here.

Reply:
We don't use this any other place?